src/components/repeat/repeat.test.tsx

```diff
@@ -0,0 +1,152 @@
+    const originalModule = jest.requireActual('../../provider/form-provider');
+    return {
+      ...originalModule,
+      useFormProviderContext: jest.fn(),
+    };
+  });
+
+  jest.mock('../../api', () => ({}));
+
+  const renderForm = async (mode: SessionMode = 'enter') => {
+    await act(async () => {
+      render(
+        <FormEngine
+          formJson={repeatingComponentTestForm as FormSchema}
+          patientUUID="8673ee4f-e2ab-4077-ba55-4980f408773e"
+          mode={mode}
+          encounterUUID={mode === 'edit' ? 'a8817ad2-ef92-46c8-bbf7-db336505027c' : null}
+        />,
+      );
+    });
+  };
+
+  const user = userEvent.setup();
+
+  const mockContext = {
+    patient: {},
+    sessionMode: 'enter',
+    formFields: repeatingComponentTestForm.pages[0].sections[0].questions,
+    methods: { getValues: jest.fn(), setValue: jest.fn() },
+    addFormField: jest.fn(),
+    formFieldAdapters: { obsGroup: { transformFieldValue: jest.fn() } },
+  };
+
+  beforeEach(() => {
+    Object.defineProperty(window, 'i18next', {
+      writable: true,
+      configurable: true,
+      value: {
+        language: 'en',
+        t: jest.fn(),
+      },
+    });
+
+    mockUsePatient.mockImplementation(() => ({
+      patient: mockPatient,
+      isLoading: false,
+      error: undefined,
+      patientUuid: mockPatient.id,
+    }));
+
+    mockUseSession.mockImplementation(() => mockSessionDataResponse.data);
+    (useFormProviderContext as jest.Mock).mockReturnValue(mockContext);
+  });
+
+  it('Should add a repeatable field instance on clicking "Add"', async () => {
+    await renderForm();
+    const addButton = screen.getByText(/add/i);
+    await user.click(addButton);
+
+    expect(mockContext.addFormField).toHaveBeenCalledTimes(1);
+    expect(mockContext.addFormField).toHaveBeenCalledWith(expect.objectContaining({ id: 'patientContact_1' }));
+  });
+
+  it('Should clone the field at origin on clicking "Add"', async () => {
+    await renderForm();
+    const addButton = screen.getByText(/add/i);
+    await user.click(addButton);
+
+    const clonedField = screen.getByLabelText(/Contact relationship/i);
+    expect(clonedField).toBeInTheDocument();
+  });
+
+  it('Should submit both the origin and its instances\' values successfully', async () => {
+    await renderForm();
+    const addButton = screen.getByText(/add/i);
+    await user.click(addButton);
+
+    const contactRelationshipField = screen.getByLabelText(/Contact relationship/i);
+    const phoneNumberField = screen.getByLabelText(/Phone/i);
+
+    await user.type(contactRelationshipField, 'Child');
+    await user.type(phoneNumberField, '123456789');
+
+    const submitButton = screen.getByText(/save/i);
+    await user.click(submitButton);
+
+    await waitFor(() => {
+      expect(mockContext.methods.getValues).toHaveBeenCalledWith(expect.objectContaining({
+        patientContactRelationship: 'Child',
+        phoneNumber: '123456789',
+      }));
```

Comment:
Can you also provide values for the instances so that we end up with a submission like:

```ts
     {
        patientContactRelationship: 'Child',
        patientContactRelationship_2: 'foo',
        phoneNumber: '123456789',
        phoneNumber_2: 'bar'
      }

```